packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
         let code = match fs::read_to_string(path.clone()) {
             Ok(s) => s,
             Err(_) => {
                 tracing::debug!("Couldn't read package file. It might intentionally not exist.");
                 // Packages don't have to use JS.
-                let fns = mv8.create_array();
-                fns.set(0, mv8::Value::Undefined)?;
-                fns.set(1, mv8::Value::Undefined)?;
-                fns.set(2, mv8::Value::Undefined)?;
+                let fns = v8::Array::new(scope, 3);
+                let undefined = v8::undefined(scope).into();
+                fns.set_index(scope, 0, undefined)
+                    .ok_or_else(|| Error::V8("Could not create Undefined value".to_string()))?;
+                fns.set_index(scope, 1, undefined)
+                    .ok_or_else(|| Error::V8("Could not create Undefined value".to_string()))?;
+                fns.set_index(scope, 2, undefined)
+                    .ok_or_else(|| Error::V8("Could not create Undefined value".to_string()))?;
+
                 return Ok(JsPackage { fns });
             }
         };
 
         // Avoid JS ReferenceError by wrapping potentially undeclared variables with `typeof`.
-        // (Double braces like `{{` are Rust's escape string for a single literal `{`.)
-        let wrapped_code = format!(
-            "((hash_util, hash_stdlib)=>{{{}
-            return [
-                typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                typeof start_sim === 'undefined' ? undefined : start_sim,
-                typeof run_task === 'undefined' ? undefined : run_task
-            ]}})",
-            code
-        );
-        let pkg: mv8::Function<'_> = mv8
-            .eval(wrapped_code)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        let args = mv8::Values::from_vec(vec![
-            embedded.hash_util.clone(),
-            embedded.hash_stdlib.clone(),
-        ]);
-
-        let fns: mv8::Array<'_> = pkg
-            .call(args)
-            .map_err(|e| Error::PackageImport(path.clone(), e.into()))?;
-        if fns.len() != 3 {
+        // Double braces like `{{` are Rust's escape string for a single literal `{`.
+        let wrapped_code = new_js_string(
+            scope,
+            &format!(
+                "((hash_util, hash_stdlib) => {{{code}
+                    return [
+                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
+                        typeof start_sim === 'undefined' ? undefined : start_sim,
+                        typeof run_task === 'undefined' ? undefined : run_task
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let pkg = v8::Script::compile(scope, wrapped_code, None)
+                .ok_or_else(|| Error::V8("Could not create wrapped code scipt".to_string()))?;
```

Comment:
```suggestion
                .ok_or_else(|| Error::V8(format!("Could not compile code for package: {name}")))?;
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2